ir/instr.cpp

```diff
@@ -2311,17 +2311,18 @@ void Alloc::print(std::ostream &os) const {
 
 StateValue Alloc::toSMT(State &s) const {
   auto [sz, np] = s[*size];
+  sz = sz.zextOrTrunc(bits_size_t);
   s.addUB(move(np));
 
   if (mul) {
     auto &[mul_e, mul_np] = s[*mul];
     s.addUB(mul_np);
-    sz = sz.zextOrTrunc(bits_size_t);
     auto m = mul_e.zextOrTrunc(bits_size_t);
     s.addUB(sz.mul_no_uoverflow(m));
     sz = sz * m;
   }
 
+  s.addUB(sz.extract(bits_size_t - 1, bits_size_t - 1) == 0);
```

Comment:
this is not correct. There's no word in the standard about overflow being UB. And now that I think about this, I don't like the existing code that a few lines above..